src/main/java/com/google/starfish/services/FavoriteNoteService.java

```diff
@@ -115,6 +128,141 @@ public long getNumFavoritesByNoteId(DataSource pool, long noteId) throws SQLExce
     }
   }
 
+  /** Gets trending notes today */
+  public Object[][] getTrendingNotesToday(DataSource pool) throws SQLException {
+    return getTrendingNotes(pool, Recency.TODAY);
+  }
+
+  /** Gets trending notes this week */
+  public Object[][] getTrendingNotesThisWeek(DataSource pool) throws SQLException {
+    return getTrendingNotes(pool, Recency.THIS_WEEK);
+  }
+
+  /** Gets trending notes this month */
+  public Object[][] getTrendingNotesThisMonth(DataSource pool) throws SQLException {
+    return getTrendingNotes(pool, Recency.THIS_MONTH);
+  }
+
+  /** Gets all-time trending notes */
+  public Object[][] getTrendingNotesAllTime(DataSource pool) throws SQLException {
+    return getTrendingNotes(pool, Recency.ALL_TIME);
+  }
+
+  /** Gets trending notes today filtered by school or course */
+  public Object[][] getTrendingNotesTodayBySchoolOrCourse(DataSource pool, String school, String course) throws SQLException {
```

Comment:
I wonder if it's really necessary to define all of these wrappers.... it might be worth it to condense this and simply call the more generic function with the specified time period from the front-end since you already have switch logic below where you can just specify it as a parameter according to the right case

Reply:
100%, I think the additional wrappers from having 2 main "wrapped" functions is overkill and you're right, since the switch logic is already here it probably just makes more sense to call the method with the enum directly. I'll make those changes.